src/quadrature/quadrature.hh

```diff
@@ -45,60 +45,90 @@
  * - introduce Quadrature point which should store point coords together with the weight in raw double[1+dim] array
  *   and just return arma object on the flys
  */
-template<unsigned int dim>
 class Quadrature {
 public:
+
+    /// Copy constructor.
+    Quadrature(const Quadrature &q);
+    
     /**
      * @brief Constructor.
      * @param n_quadrature_points Number of quadrature points to be allocated.
      */
-    Quadrature(const unsigned int n_quadrature_points = 0);
-
-    /// Copy constructor.
-    Quadrature(const Quadrature<dim> &q);
+    Quadrature(unsigned int dimension, const unsigned int n_quadrature_points = 0);
 
     /** @brief Constructor from quadrature of lower dimension (e.g. for side integration).
-     * @param sub_quadrature lower dimnesional (dim-1) quadrature
+     * @param sub_quadrature lower dimensional (dim-1) quadrature
      * @param sid local index of side
      * @param pid index of permutation of nodes on given side
      */
-    Quadrature(const Quadrature<dim-1> &sub_quadrature, unsigned int sid, unsigned int pid);
+//     template<unsigned int quad_dim>
+//     explicit Quadrature(const Quadrature &sub_quadrature, unsigned int sid, unsigned int pid);
     
     /// Virtual destructor.
-    virtual ~Quadrature();
+    virtual ~Quadrature() {};
+    
+    inline unsigned int dim() const
+    { return dim_; }
 
     /**
      * @brief Modify the number of quadrature points.
      * @param n_q_points New number of quadrature points.
      */
-    void resize(const unsigned int n_q_points);
+    inline void resize(const unsigned int n_q_points)
+    {
+        quadrature_points.resize(n_q_points);
+        weights.resize(n_q_points, 0);
+    }
 
     /// Returns number of quadrature points.
-    const unsigned int size() const;
+    inline const unsigned int size() const
+    { return weights.size(); }
 
     /// Returns the <tt>i</tt>th quadrature point.
-    arma::vec::fixed<dim> point(const unsigned int i) const;
+    template<unsigned int point_dim>
+    inline arma::vec::fixed<point_dim> point(const unsigned int i) const
```

Comment:
We can simplify the interface by returning directly the Mat object returned by get<point_dim>(i).
This way we can assign to the point and we can drop the set_point method. Same for the weights.

Reply:
Fixed.